third_party/move/tools/move-cli/src/base/test_validation.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) The Diem Core Contributors
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module manages validation of the unit tests, in addition to standard compiler
+//! checking.
+
+use move_model::model::GlobalEnv;
+use once_cell::sync::Lazy;
+use std::sync::Mutex;
+
+static VALIDATION_HOOK: Lazy<Mutex<Option<Box<dyn Fn(&GlobalEnv) + Send + Sync>>>> =
```

Comment:
This changeable global state is pretty horrible.  Can you avoid it?

We already have problems building/testing in parallel, and this just adds another reason why tests might not behave well.

Can't this info be conveyed in a parameter to the tests somehow?

Reply:
I know it is horrible, but as said in the PR description, it is replicating an existing hack as [seen here](https://github.com/aptos-labs/aptos-core/blob/a1ae60836f987cacb6ed3f731e585df7c684c67e/third_party/move/tools/move-unit-test/src/extensions.rs#L19). One big problem with extending the UT package taking an external trait is that all configuration options have to be serializable, and so you can't just add a `dyn` to this data. I think this serializability is in turn needed for some functionality of the `move-cli` -- dead code, put not easy to remove. This is what I meant with the larger refactoring which need to be done first to do this here right. So this is a shortcut until we have cycles for the bigger thing.

Reply:
Ok for now.